espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/InvocationOperation.java

```diff
@@ -0,0 +1,376 @@
+    private final Class<?>[] argumentTypes;
+
+    public InvocationOperation(String methodName, List<?> arguments) {
+        this.methodName = methodName;
+        this.arguments = arguments;
+        this.argumentTypes = parseArgumentTypes(arguments);
+    }
+
+
+    public Object apply(final Object o) throws Exception {
+        Log.d("Backdoor Method", getName());
+        Log.d("Invoking on ", o.toString());
+        final AtomicReference<Object> ref = new AtomicReference<Object>();
+        final AtomicReference<Exception> refEx = new AtomicReference<Exception>();
+
+        InstrumentationRegistry.getInstrumentation().runOnMainSync(new Runnable() {
+            @Override
+            public void run() {
+                MethodWithArguments method = findCompatibleMethod(o);
+
+                if (method == null) {
+                    StringBuilder stringBuilder = new StringBuilder("No such method found: ");
+                    stringBuilder.append(InvocationOperation.this.methodName);
+                    stringBuilder.append("(");
+                    int length = InvocationOperation.this.arguments.size();
+
+                    for (int i = 0; i < length; i++) {
+                        Object argument = InvocationOperation.this.arguments.get(i);
+
+                        if (i != 0) {
+                            stringBuilder.append(", ");
+                        }
+
+                        String simpleName;
+
+                        if (argument == null) {
+                            simpleName = "null";
+                        } else {
+                            simpleName = argument.getClass().getSimpleName();
+                        }
+
+                        stringBuilder.append("[").append(simpleName).append("]");
+                    }
+
+                    stringBuilder.append(")");
+
+                    ref.set(BackdoorResultVoid.instance
+                            .asMap(InvocationOperation.this.methodName, o, stringBuilder.toString()));
+                    return;
+                }
+
+                Object result;
+
+                try {
+                    if (method.getMethod().getReturnType().equals(Void.TYPE)) {
+                        method.invoke(o);
+                        result = "<VOID>";
+                    } else {
+                        result = method.invoke(o);
+                    }
+
+                    ref.set(result);
+                    return;
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    refEx.set(e);
+                    return;
+                }
+            }
+        });
+
+        if (refEx.get() != null) {
+            throw refEx.get();
+        }
+
+        return ref.get();
+    }
+
+    public MethodWithArguments findCompatibleMethod(Object object) {
+        return findCompatibleMethod(object.getClass());
+    }
+
+    public MethodWithArguments findCompatibleMethod(Class<?> forClass) {
```

Comment:
the method should be marked as `Nullable`

Reply:
added